flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -260,17 +258,28 @@ protected StreamTask(
 			@Nullable TimerService timerService,
 			Thread.UncaughtExceptionHandler uncaughtExceptionHandler,
 			StreamTaskActionExecutor actionExecutor,
-			TaskMailbox mailbox) {
+			TaskMailbox mailbox) throws Exception {
 
 		super(environment);
 
-		this.timerService = timerService;
-		this.uncaughtExceptionHandler = Preconditions.checkNotNull(uncaughtExceptionHandler);
 		this.configuration = new StreamConfig(getTaskConfiguration());
 		this.recordWriter = createRecordWriterDelegate(configuration, environment);
 		this.actionExecutor = Preconditions.checkNotNull(actionExecutor);
 		this.mailboxProcessor = new MailboxProcessor(this::processInput, mailbox, actionExecutor);
 		this.asyncExceptionHandler = new StreamTaskAsyncExceptionHandler(environment);
+		this.asyncOperationsThreadPool = Executors.newCachedThreadPool(
+			new ExecutorThreadFactory("AsyncOperations", uncaughtExceptionHandler));
+
+		this.stateBackend = createStateBackend();
```

Comment:
Should we move the appropriate line in `Task` higher:
```executingThread.setContextClassLoader(userCodeClassLoader);```
?
Now (in master) it's before `invoke()`; and it looks like it should be moved before 
```invokable = loadAndInstantiateInvokable(userCodeClassLoader, nameOfInvokableClass, env);```

Reply:
`executingThread.setContextClassLoader(userCodeClassLoader)` was actually done twice both before `loadAndInstantiateInvokable` and `invoke()`. I am not quite sure why we need to set twice before. 

Anyway it is not related to my PR change, so I do not try to figure it out ATM. WDYT?

Reply:
Yes, you are right, it's set twice :)